scipy/spatial/transform/rotation.py

```diff
@@ -417,5 +533,74 @@ def from_euler(cls, seq, angles, degrees=False):
             raise ValueError("Expected angles to have shape (num_rotations, "
                              "num_axes), got {}.".format(angles.shape))
 
-        quat = _elementary_quat_compose(seq, angles, intrinsic)
+        quat = elementary_quat_compose(seq, angles, intrinsic)
         return cls(quat[0] if is_single else quat, normalized=True)
+
+    def as_euler(self, seq, degrees=False):
+        """Return the Euler angles representation of the Rotation.
+
+        This function returns a numpy.ndarray of shape (N, 3) or (3,) depending
+        on how the object was initialized. The algorithm presented in [2]_ has
+        been adapted for our use to extract Euler angles. The paper presents an
+        algorithm for extracting Euler angles from transformation matrices, as
+        opposed to rotation matrices. Thus, the matrix representation used in
+        the paper is a transpose of the direction cosine matrix representation
+        returned by the `as_dcm` function.
+
+        The returned angles are in the range:
+
+            - First angle belongs to [-180, 180] degrees (both inclusive)
+            - Third angle belongs to [-180, 180] degrees (both inclusive)
+            - Second angle belongs to:
+
+                - [-90, 90] degrees if all axes are unique (like xyz)
+                - [0, 180] degrees if first and third axes are same (like zxz)
+
+        Euler angles suffer from the problem of gimbal lock. It occurs when,
```

Comment:
Add a reference to here https://en.wikipedia.org/wiki/Gimbal_lock#In_applied_mathematics

I believe we can omit the explanation and rely on the reference. However, I suggest to write ("In this case the third angle is set to zero. Note however that the returned angles still represent the correct rotation".